packages/build/src/buildbot_client/main.js

```diff
@@ -0,0 +1,76 @@
+const net = require('net')
+
+const { addErrorInfo } = require('../error/info')
+
+const startBuildbotClient = function(buildbotServerSocket) {
+  return new Promise((resolve, reject) => {
+    try {
+      const client = net.createConnection(buildbotServerSocket)
+      client.on('connect', () => resolve(client))
```

Comment:
The TCP socket can emit [`error` events](https://nodejs.org/api/net.html#net_event_error_1), for example when there is a TCP connection problem. 
When listening to either the `connect` or `data` event, we should also listen for any `error` event, and propagate the error.

One quick way to do it would be to use [`p-event`](https://github.com/sindresorhus/p-event), which promisifies events and rejects the `Promise` on any `error` event [by default](https://github.com/sindresorhus/p-event#rejectionevents). This would also remove the need for `new Promise()`. For example:

```js
const startBuildbotClient = async function(buildbotServerSocket) {
  try {
    const client = net.createConnection(buildbotServerSocket)
    // This will throw if `client` emits `error`
    await pEvent(client, 'connect')
    return client
  } catch (error) {
    addErrorInfo(error, { type: 'buildbotClientStartup' })
    throw error
  }
}

const getNextParsedResponsePromise = async function(client) {
  // This will throw if `client` emits `error`
  const data = await pEvent(client, 'data')
  try {
    return JSON.parse(data)
  } catch (error) { 
    // Handle error
  }
}
```